2996_reflection/reflection.md

```diff
@@ -5267,15 +5477,15 @@ consteval bool has_identifier(info r);
 
 [#]{.pnum} *Returns*:
 
-* [#.#]{.pnum} If `r` represents a class type `$C$`, then `true` when either `$C$` has a typedef name for linkage purposes ([dcl.typedef]) or the `$class-name$` introduced by the declaration of `$C$` is an identifier. Otherwise, `false`.
-* [#.#]{.pnum} Otherwise, if `r` represents an unnamed entity, then `false`.
+* [#.#]{.pnum} If `r` is an unnamed entity other than a class that has a typedef name for linkage purposes ([dcl.typedef]{.sref}), then `false`.
+* [#.#]{.pnum} Otherwise, if `r` represents a class type `$C$`, then `true` when either the `$class-name$` of `$C$` is an identifier or `$C$` has a typedef name for linkage purposes. Otherwise, `false`.
 * [#.#]{.pnum} Otherwise, if `r` represents a function, then `true` if the function is not a function template specialization, constructor, destructor, operator function, or conversion function. Otherwise, `false`.
 * [#.#]{.pnum} Otherwise, if `r` represents a function template, then `true` if `r` does not represent a constructor template, operator function template, or conversion function template. Otherwise, `false`.
-* [#.#]{.pnum} Otherwise, if `r` represents a `$typedef-name$`, then `true` when the `$typedef-name$` is an identifier. Otherwise, `false`.
+* [#.#]{.pnum} Otherwise, if `r` represents a type alias, then `true` when the type alias is not a specialization of an alias template. Otherwise, `false`.
 * [#.#]{.pnum} Otherwise, if `r` represents a variable, then `true` if `r` does not represent a variable template specialization. Otherwise, `false`.
-* [#.#]{.pnum} Otherwise, if `r` represents a structured binding, enumerator, non-static data member, template, namespace, or `$namespace-alias$`, then `true`. Otherwise, `false`.
-* [#.#]{.pnum} Otherwise, if `r` represents a base class specifier, then `true` if `has_identifier(type_of(r))`. Otherwise, `false`.
-* [#.#]{.pnum} Otherwise if `r` represents a description of a declaration of a non-static data member, then letting `$o$` be a `data_member_options` value such that `data_member_spec(type_of(r), $o$) == $r$`, then `true` if `$o$.name` contains a value. Otherwise, `false`.
+* [#.#]{.pnum} Otherwise, if `r` represents a structured binding, enumerator, non-static data member, template, namespace, or namespace alias, then `true`. Otherwise, `false`.
+* [#.#]{.pnum} Otherwise, if `r` represents a direct base class relationship, then `true` if `has_identifier(type_of(r))`. Otherwise, `false`.
+* [#.#]{.pnum} Otherwise if `r` represents a data member description (`$T$`, `$N$`, `$A$`, `$W$`, `$NU$`) ([class.mem.general]{.sref}), then `true` if `$N$` is not `-1`. Otherwise, `false`.
```

Comment:
```suggestion
* [#.#]{.pnum} Otherwise if `r` represents a data member description (`$T$`, `$N$`, `$A$`, `$W$`, `$NU$`) ([class.mem.general]{.sref}), then `$N$ != -1`.
```
Although it's weird here that `$N$` is either a string or an integer?

Reply:
It is, but it's exposition-only bullshit. Maybe it's fine.